feat(toolkit-examples): dmail and web3mail integration guide

Nkovaturient opened the pull request:
## Description
- addresses sub-issue #5 

## Dmail Integration + Web3Mail Integration 
### `integration-guide` folder encompasses `Auth` , `Space Suite`, and `Upload Tool `integration in dmail and web3mail services
- simple dapp demonstarting Storacha Toolkit's Auth Component ( `useStorachaAuth` and `StorachaAuth`), Space Suites, and Upload Tool integration examples site
- Simplified import paths for easier local development

> Both examples now demonstrate proper integration of Storacha Console Toolkit components with clear separation of concerns and improved developer experience.

---

<img width="1703" height="985" alt="Screenshot 2025-12-06 at 9 22 22 PM" src="https://github.com/user-attachments/assets/dac122c9-a07c-4209-b446-f5115d553ef9" />

Nkovaturient commented:
@travis @Dhruv-Varshney-developer kindly review it.

Nkovaturient commented:
- Streamlined auth flow and improvised working of StorachaAuth in web3mail example demo


https://github.com/user-attachments/assets/7958bb10-eda7-43d6-8229-62fe904e9d97

Dhruv-Varshney-developer commented:
Thanks for the work here @Nkovaturient !
Assigning @alikhere for reviews here. 
Feel free to skip our team's review of individual PRs on this project.
Please review PRs among yourselves. 
Feel free to tag me around if there's a major update or a blocker here.

Nkovaturient commented:
@alikhere Kindly review it on these latest changes.
 
###  `packages/react/src/components/upload/UploadTool.tsx:`
-  fixed hook-order bug in `UploadToolTypeSelector` (hook before conditional), removed duplicate setUploadType, added stable keys when using renderOption to silence React warnings.

### Web3Mail & Dmail: 
- **added UploadTool UI  and Space Suite components to showcase their workings and integration process in above services.**
 - integrated quick upload + advanced upload trigger

| Space creation UI | Upload Tool |
| --- | --- |
| <img width="550" height="850" alt="Screenshot 2025-12-19 at 2 58 29 PM" src="https://github.com/user-attachments/assets/4f461c0b-5ce1-4040-a632-a37d16266fb6" /> | <img width="550" height="850" alt="Screenshot 2025-12-19 at 3 58 56 PM" src="https://github.com/user-attachments/assets/2ffc81d4-a8c9-47b1-9ece-18d2d53aa568" /> |


<img width="1681" height="832" alt="advanced-upload" src="https://github.com/user-attachments/assets/7c1d0eba-72e0-4054-a22b-77d389624bd0" />


## To Fix:
| Provider err after new Private/Public Space addition | Invocation err while uploading |
| --- | --- |
| <img width="1411" height="830" alt="provider err" src="https://github.com/user-attachments/assets/fb9807f0-f0bf-45c2-a3a2-f8652be92787" /> | <img width="1476" height="796" alt="invocation err" src="https://github.com/user-attachments/assets/d6d4cf6e-efe5-4ea9-80da-93880ad5215c" /> |

Nkovaturient commented:
Hey @alikhere 
I have improved the upload dropzone clickability area in both the dmail and web3mail upload tool components.
I have also checked for any other issues. Let me know in case I missed any.
I believe, so far, it's good to merge.
